_stbt/motion.py

```diff
@@ -0,0 +1,348 @@
+from collections import deque
+
+import cv2
+import numpy
+
+from .config import ConfigurationError, get_config
+from .imgutils import (_frame_repr, _image_region, _ImageFromUser, _load_image,
+                       crop)
+from .logging import debug, draw_on, ImageLogger
+from .types import Region, UITestFailure
+
+
+def detect_motion(timeout_secs=10, noise_threshold=None, mask=None,
+                  region=Region.ALL, frames=None):
+    """Generator that yields a sequence of one `MotionResult` for each frame
+    processed from the device-under-test's video stream.
+
+    The `MotionResult` indicates whether any motion was detected -- that is,
+    any difference between two consecutive frames.
+
+    :type timeout_secs: int or float or None
+    :param timeout_secs:
+        A timeout in seconds. After this timeout the iterator will be exhausted.
+        Thas is, a ``for`` loop like ``for m in detect_motion(timeout_secs=10)``
+        will terminate after 10 seconds. If ``timeout_secs`` is ``None`` then
+        the iterator will yield frames forever. Note that you can stop
+        iterating (for example with ``break``) at any time.
+
+    :param float noise_threshold:
+        The amount of noise to ignore. This is only useful with noisy analogue
+        video sources. Valid values range from 0 (all differences are
+        considered noise; a value of 0 will never report motion) to 1.0 (any
+        difference is considered motion).
+
+        This defaults to 0.84. You can override the global default value by
+        setting ``noise_threshold`` in the ``[motion]`` section of
+        :ref:`.stbt.conf`.
+
+    :type mask: str or `numpy.ndarray`
+    :param mask:
+        A black & white image that specifies which part of the image to search
+        for motion. White pixels select the area to analyse; black pixels select
+        the area to ignore. The mask must be the same size as the video frame.
+
+        This can be a string (a filename that will be resolved as per
+        `load_image`) or a single-channel image in OpenCV format.
+
+    :type region: `Region`
+    :param region:
+        Only analyze the specified region of the video frame.
+
+        If you specify both ``region`` and ``mask``, the mask must be the same
+        size as the region.
+
+    :type frames: Iterator[stbt.Frame]
```

Comment:
This is actually `Iterator[(stbt.Frame, int)]`.

Reply:
Fixed